lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -73,6 +92,25 @@ mkJormungandrLayer mgr baseUrl = JormungandrLayer
     { getTipId = ExceptT $ do
         let ctx = safeLink api (Proxy @GetTipId)
         run cGetTipId >>= defaultHandler ctx
+    , getBlock = \blockId -> ExceptT $ do
+        run (cGetBlock blockId)  >>= \case
+            Left (FailureResponse e) | responseStatusCode e == status404 ->
+              return $ Right Nothing
+            x -> do
+                let ctx = safeLink api (Proxy @GetBlock) blockId
+                right Just <$> defaultHandler ctx x
+    , getDescendantIds = \parentId count -> ExceptT $ do
+        run (cGetBlockDescendantIds parentId (Just count))  >>= \case
+            Left (FailureResponse e) | responseStatusCode e == status404 ->
+              return $ Left ErrGetDescendantsParentNotFound
+            x -> do
+                let ctx = safeLink
+                        api
+                        (Proxy @GetBlockDescendantIds)
+                        parentId
+                        (Just count)
+                left ErrGetDescendantsNetworkUnreachable <$> defaultHandler ctx x
```

Comment:
I ended up with two styles here. Let's say it was deliberate. If `getBlock` 404s we get `Nothing`, and if `getDescendantIds` 404s we get `ErrGetDescendantsParentNotFound `.

Which one should we go with?

Reply:
Hmmm. It seems more logical to have `getBlock` returns `Block` and, fail with a `BlockNotFound` or something like this than returning a `Maybe Block`.
It's all about the usage of that API; here, we do indeed expect a block when we call `getBlock` (otherwise, where do we take this id from?). If there's no block, caller can't do much and most likely, fail. I'd suggest to follow an API a bit closer to what exists in the `http-bridge`:

```hs
    { getBlock
        :: Hash "BlockHeader" -> ExceptT ErrNetworkUnreachable m Block
    , getEpoch
        :: Word64 -> ExceptT ErrNetworkUnreachable m [Block]
    , getNetworkTip
        :: ExceptT ErrNetworkTip m (Hash "BlockHeader", BlockHeader)
    , postSignedTx
        :: (Tx, [TxWitness]) -> ExceptT ErrPostTx m ()
```

Note that the bridge is even more severe on the error and simply throws if there's no block. Having an error `ErrGetBlock` that captures both the network failure and the absence of block would be nicer (in Jormungandr)